doc/new_boilerplate_migration.md

```diff
@@ -803,20 +816,52 @@ UiFactory<FooProps> createFooHoc(UiFactory otherFactory) {
   Object closureVariable; 
   // ... 
 
-  final FooHoc = uiFunctionComponent<FooProps>(
+  final FooHoc = uiFunction<FooProps>(
     (props) { 
       return otherFactory()( 
         Dom.div()('closureVariable: ${closureVariable}'), 
         Dom.div()('prop foo: ${props.foo}'), 
       );   
     },
-    $fooPropsConfig, // ignore: undefined_identifier 
+    FunctionComponentConfig(
+        displayName: 'FooHoc',
+        propsFactory: PropsFactory.fromUiFactory(Foo),
+    ),
   ); 
 
   return FooHoc; 
 } 
 ```
 
+#### With forwardRef
+
+```dart
+mixin FooProps on UiProps {
+  Ref forwardedRef;
+  Function doSomething;
+}
+
+UiFactory<FooProps> Foo = forwardRef<FooProps>((props, ref) {
+  return (_Foo()
+    ..forwardedRef = ref
+    ..doSomething = props.doSomething
+  )();
+})(_Foo);
+
+UiFactory<FooProps> _Foo = uiFunction(
+  (props) {
+    return Fragment()(
+      Dom.div()('Some text.'),
+      (Dom.button()
+        ..ref = props.forwardedRef 
+        ..onClick = props.doSomething
+      )('Click me!'),
+    );
+  },
+  $_FooConfig, // ignore: undefined_identifier
+);
```

Comment:
Can we update this to utilize the new syntax that will be added in https://jira.atl.workiva.net/browse/CPLAT-11722 ?
```suggestion
UiFactory<FooProps> Foo = uiForwardRef(
  (props, ref) {
    return Fragment()(
      Dom.div()('Some text.'),
      (Dom.button()
        ..ref = ref
        ..onClick = props.doSomething
      )('Click me!'),
    );
  },
  $_FooConfig, // ignore: undefined_identifier
);
```